### PR TITLE
Yti 1605 reusable components

### PR DESCRIPTION
**Pull request by topirissanen-atoz**
Updated components to be more general and fixed styles.

- Filter has been broken down into smaller components in filter-folder
- Updated terminology-search-slice to add needed functionality for Filter to work on Search-page.
- Removed bunch of old code from /terminology-search and /vocabulary - both now only have slice files.

Search-page without any additional filters.
image

Search-page with additional filters.
image

Terminology-page without any additional filters.
image

Terminology-page with additional filters.
image

Terminology-page information.
image

### CODE REVIEW

File: src/common/components/filter/filter.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+  if (type === 'vocabulary' && 'showBy' in filter) {
+    return (
+      <FilterWrapper>
+        <Header>
+          {t('vocabulary-filter-filter-list')}
+        </Header>
+
+        {/* If filter has any value 'checked' Remove-component is displayed. */}
+        {JSON.stringify(filter) !== JSON.stringify(vocabularyEmptyState.filter) &&
```
Comment (jukkah) [inline src/common/components/filter/filter.tsx:46]: Note that `JSON.stringify({ a: 1, b: 1 }) !== JSON.stringify({ b: 1, a: 1 })`. Is it possible that these object were constructed in different order?

File: src/common/components/filter/filter.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    return (
+      <FilterWrapper>
+        <Header>
+          {t('vocabulary-filter-filter-list')}
+        </Header>
+
+        {/* If filter has any value 'checked' Remove-component is displayed. */}
+        {
+          (
+            JSON.stringify({ ...filter, infoDomains: {} }) !== JSON.stringify(initialState.filter)
```
Comment (jukkah) [inline src/common/components/filter/filter.tsx:90]: Same here.

File: src/common/components/search-results/search-results.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+              <Card key={`search-result-${idx}`}>
+                <CardContributor>
+                  {terminology.contributors[0].label[i18n.language]}
+                </CardContributor>
+
+                <CardTitle variant='h2'>
+                  <Link passHref href={'/terminology/' + terminology.id}>
+                    <CardTitleLink href=''>
+                      <CardTitleIcon icon='registers' />
+                      <span dangerouslySetInnerHTML={{
```
Comment (jukkah) [inline src/common/components/search-results/search-results.tsx:52]: Why dangerouslySetInnerHTML? I can't see why normal conditional rendering would not work.

**Review by jukkah — CHANGES_REQUESTED** [review]

File: src/modules/terminology-search/index.tsx [hunk 4]
```diff
@@ -19,46 +18,44 @@ import Pagination from '../../common/components/pagination/pagination';
...
-            (filteredData && filteredData !== null)
+            (data && data !== null)
               ?
               <PaginationWrapper>
                 <Pagination
-                  data={filteredData}
+                  data={data}
                   dispatch={dispatch}
-                  isSmall={false}
+                  isSmall={breakPoints.isSmall}
```
Comment (jukkah) [inline src/modules/terminology-search/index.tsx:58]: This could also be moved to Pagination component itself.

**Review by jukkah — APPROVED** [review]

**Review by jukkah — APPROVED** [review]

File: src/common/components/filter/checkbox-area.tsx [hunk 5]
```diff
@@ -23,7 +23,7 @@ export default function CheckboxArea({ data, filter, setFilter, title, type }: C
   const dispatch = useStoreDispatch();
 
   const handleCheckbox = (s: string | InfoDProp) => {
-    let retVal;
+    let retVal: CheckboxProps['filter'] | undefined;
```
Comment (urhot) [inline src/common/components/filter/checkbox-area.tsx:26]: I think null would be more appropriate here:

```ts
let retVal: CeckboxProps['filter'] | null = null;
```

**Pull request merged**